drivers/flash/Kconfig.sam_qspi

```diff
@@ -0,0 +1,12 @@
+# Copyright (c) 2025 Microchip Technology Inc.
+# SPDX-License-Identifier: Apache-2.0
+
+config FLASH_SAM_QSPI
+	bool "Microchip Quad SPI Flash driver"
+	default y
+	select FLASH_HAS_DRIVER_ENABLED
+	select FLASH_HAS_EXPLICIT_ERASE
+	select FLASH_JESD216
+	select DMA
```

Comment:
The DMA driver is always selected but the usage is optional in devicetree.
Why customer would use the QSPI without DMA ?
Shall we "force" user to always define the DMA?
Can user select different DMA channels or those are fixed ?